lib/driver.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+
+  attr_reader :name, :vin, :trips, :status
+  
+  def initialize(id:, name:, vin:, status: :AVAILABLE, trips: nil)
+    super(id)
+
+    @name = name
+    @vin = vin
+    @status = status.to_sym
+    @trips = trips || []
+  
+    vin_pattern = /^[A-Z0-9]{17}$/i
+    if (vin_pattern =~ @vin) == nil
+      raise ArgumentError, "Invalid VIN"
```

Comment:
It's helpful to include the bad argument in the message for your `ArgumentError`s:

```suggestion
      raise ArgumentError, "Invalid VIN: #{@vin}"
```